plugins/auth/login.js

```diff
@@ -1,97 +1,145 @@
 'use strict';
 
 const boom = require('boom');
-const urlLib = require('url');
+const uuid = require('uuid/v4');
 
 /**
- * Login to Screwdriver API
- * @method login
- * @param  {Object}      config           Configuration from the user
- * @param  {Array}       config.whitelist List of allowed users to the API
- * @return {Object}                       Hapi Plugin Route
+ * Add a guest route for those who want to be in read-only mode
+ * @method addGuestRoute
+ * @param  {Hapi}     server                        Hapi Server
+ * @param  {Object}   config                        Configuration object
+ * @param  {Boolean}  config.allowGuestAccess       Letting users browse your system
+ * @return Array                                    List of new routes
  */
-module.exports = config => ({
-    method: ['GET', 'POST'],
-    path: config.scmContext ? `/auth/login/${config.scmContext}/{web?}` : '/auth/login/{web?}',
-    config: {
-        description: 'Login using oauth',
-        notes: 'Authenticate user with oauth provider',
-        tags: ['api', 'login'],
-        auth: config.auth,
-        handler: (request, reply) => {
-            const pathNames = request.path.split('/');
-            const scmContext = pathNames[pathNames.indexOf('login') + 1];
-
-            // Redirect to the default login path if request path doesn't have a context
-            if (!scmContext || scmContext === 'web') {
-                const defaultContext = request.server.root.app.userFactory.scm.getScmContexts()[0];
-                const prefix = request.route.realm.modifiers.route.prefix;
-                let pathName = `${prefix}/auth/login/${defaultContext}`;
-
-                if (request.params.web) {
-                    pathName += '/web';
+function addGuestRoute(server, config) {
+    return [{
+        method: ['GET'],
+        path: '/auth/login/guest/{web?}',
+        config: {
+            description: 'Login as an guest user',
+            notes: 'Authenticate an guest user',
+            tags: ['api', 'login'],
+            auth: null,
+            handler: (request, reply) => {
+                // Check if guest is allowed to login
+                if (!config.allowGuestAccess) {
+                    return reply(boom.forbidden('Guest users are not allowed access'));
                 }
 
-                const location = urlLib.format({
-                    host: request.headers.host,
-                    port: request.headers.port,
-                    protocol: request.server.info.protocol,
-                    pathname: pathName
-                });
+                const username = `guest/${uuid()}`;
+                const profile = request.server.plugins.auth.generateProfile(
+                    username, 'guest', ['user', 'guest'], {}
```

Comment:
So guest user has 2 scopes: `user` and `guest`?

Reply:
Yes, so a guest is a user, but is also a guest.